hudi-client/hudi-client-common/src/main/java/org/apache/hudi/utils/HiveHasher.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.utils;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import sun.misc.Unsafe;
+
+public class HiveHasher {
```

Comment:
Why do we specifically need the hashused by hive?

Reply:
This will be Something we will have to now maintain for a long time. I Would prefer we used Some simpler performant hashing aIgorithm that came with the Jvm

Reply:
or make the hash method an interface and configurable? Flink uses murmurhash. We might extend this for different use cases later.